src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -614,29 +621,27 @@ private HaxeClassResolveResult resolveHaxeClassInternal() {
     if(isType(resolve, HaxeParameter.class)) {
       // check if  type parameters has multiple constraints and try to unify
       HaxeTypeTag tag = ((HaxeParameter)resolve).getTypeTag();
-      String typeName = tag != null ? tag.getTypeOrAnonymous().getText() : null;
+      String typeName = tag != null  && tag.getTypeOrAnonymous() != null ? tag.getTypeOrAnonymous().getText() : null;
       PsiElement parameterList = resolve.getParent();
 
       if(parameterList != null && parameterList.getParent() instanceof HaxeMethodDeclaration ) {
         HaxeMethodDeclaration method = (HaxeMethodDeclaration)parameterList.getParent();
-        if(method != null) {
-          HaxeGenericParam genericParam = method.getGenericParam();
-          List<HaxeGenericListPart> partList =genericParam != null ? genericParam.getGenericListPartList() : null;
-          if(partList!= null) {
-            Optional<HaxeGenericListPart> match = partList.stream().filter(part -> Objects.equals(typeName, part.getName())).findFirst();
-            if(match.isPresent()) {
-              HaxeGenericListPart listPart = match.get();
-              HaxeTypeList list = listPart.getTypeList();
-              if(list != null) {
-                list.getTypeListPartList();
-                List<HaxeType> classReferences = list.getTypeListPartList().stream()
-                  .map(part -> part.getTypeOrAnonymous() == null ? null : part.getTypeOrAnonymous().getType())
-                  .filter(Objects::nonNull)
-                  .collect(Collectors.toList());
-
-                HaxeTypeParameterMultiType constraint = new HaxeTypeParameterMultiType(listPart.getContext().getNode(), classReferences);
-                return HaxeClassResolveResult.create(constraint);
-              }
+        HaxeGenericParam genericParam = method.getGenericParam();
+        List<HaxeGenericListPart> partList =genericParam != null ? genericParam.getGenericListPartList() : null;
+        if(partList!= null) {
+          Optional<HaxeGenericListPart> match = partList.stream().filter(part -> Objects.equals(typeName, part.getName())).findFirst();
```

Comment:
Oh... Let's ditch the streams in this function, which is called a LOT.  Lines 632 and 638.